Src/Particle/AMReX_ParticleContainerI.H

```diff
@@ -646,6 +646,36 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>::Re
   }
 }
 
+struct TransformerVirt
+{
+    /**
+       \brief This copies the particle but replaces the id with VirtualParticleID
+       *
+     **/
+    TransformerVirt ()
+    {}
+
+    template <typename DstData, typename SrcData>
+    AMREX_GPU_HOST_DEVICE
+    void operator() (DstData& dst, const SrcData& src,
+                     int src_i, int dst_i) const noexcept
+    {
+        dst.m_aos[dst_i] = src.m_aos[src_i];
```

Comment:
I think you can save some code here by using the `copyParticle` function in `AMReX_ParticleTransformation.H`.

Reply:
Pushed a commit which does this

Reply:
Nice - not needed for this PR, but you could also update the ghost one to do the same thing.

Reply:
I did include update as well